presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -172,8 +193,16 @@ protected Node visitQuerySpecification(QuerySpecification node, Void context)
         if (!removablePrefix.isPresent()) {
             removablePrefix = Optional.of(new Identifier(baseTable.getName().toString()));
         }
-        if (materializedViewInfo.getGroupBy().isPresent() && !node.getGroupBy().isPresent()) {
-            throw new IllegalStateException("Query with no groupBy clause is not rewritable by materialized view with groupBy clause");
+        if (node.getGroupBy().isPresent()) {
```

Comment:
This logic can be moved to visitGroupBy, right?

Reply:
I moved the part here is because we need to know which expressions are present in `GROUP BY` before visiting `SELECT`.
If a selected column is in `GROUP BY` of materialized view, this query might be eligible for optimization, otherwise it cannot.

This is based on the observation of the example:
Materialized View definition:
```SQL
CREATE MATERIALIZED VIEW mv
SELECT SUM(a) AS a, b FROM base GROUP BY b
```

The following query can be optimized:
```SQL
SELECT b FROM base GROUP BY b
```

While the following one can not:
```SQL
SELECT b FROM base
```